examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+
+
+### Retriever Microservice
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-retriever-usvc 7000:7000
+```
+Follow the below steps in a different terminal.
+
+To consume the retriever microservice, you need to generate a mock embedding
+vector by Python script. The length of embedding vector is determined by the
+embedding model. Here we use the
+model EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5", which vector size is 768.
+
+Check the vector dimension of your embedding model and set
+`your_embedding` dimension equal to it.
+
+```
+export your_embedding=$(python3 -c "import random; embedding = [random.uniform(-1, 1) for _ in range(768)]; print(embedding)")
+
+curl http://localhost:7000/v1/retrieval \
+  -X POST \
+  -d "{\"text\":\"test\",\"embedding\":${your_embedding}}" \
+  -H 'Content-Type: application/json'
+
+```
+The output of the retriever microservice comprises of the a unique id for the
+request, initial query or the input to the retrieval microservice, a list of top
+`n` retrieved documents relevant to the input query, and top_n where n refers to
+the number of documents to be returned.
+
+The output is retrieved text that relevant to the input data:
+```
+{"id":"27210945c7c6c054fa7355bdd4cde818","retrieved_docs":[{"id":"0c1dd04b31ab87a5468d65f98e33a9f6","text":"Company: Nike. financial instruments are subject to master netting arrangements that allow for the offset of assets and liabilities in the event of default or early termination of the contract.\nAny amounts of cash collateral received related to these instruments associated with the Company's credit-related contingent features are recorded in Cash and\nequivalents and Accrued liabilities, the latter of which would further offset against the Company's derivative asset balance. Any amounts of cash collateral posted related\nto these instruments associated with the Company's credit-related contingent features are recorded in Prepaid expenses and other current assets, which would further\noffset against the Company's derivative liability balance. Cash collateral received or posted related to the Company's credit-related contingent features is presented in the\nCash provided by operations component of the Consolidated Statements of Cash Flows. The Company does not recognize amounts of non-cash collateral received, such\nas securities, on the Consolidated Balance Sheets. For further information related to credit risk, refer to Note 12 — Risk Management and Derivatives.\n2023 FORM 10-K 68Table of Contents\nThe following tables present information about the Company's derivative assets and liabilities measured at fair value on a recurring basis and indicate the level in the fair\nvalue hierarchy in which the Company classifies the fair value measurement:\nMAY 31, 2023\nDERIVATIVE ASSETS\nDERIVATIVE LIABILITIES"},{"id":"1d742199fb1a86aa8c3f7bcd580d94af","text": ... }
```

Comment:
With the uploaded document being about OPEA, output being about Nike fiscals seems unlikely.

Reply:
@eero-t captured and added OPEA document